python/ray/data/_internal/plan.py

```diff
@@ -216,7 +216,7 @@ def get_plan_as_string(self, classname: str) -> str:
         if dataset_blocks is None:
             num_blocks = "?"
         else:
-            num_blocks = dataset_blocks.initial_num_blocks()
+            num_blocks = dataset_blocks.estimated_num_blocks()
```

Comment:
I guess this is just for debugging? I think it's a big confusing that this num_blocks might not be right.

Reply:
Yeah. This was always an estimate, since block splitting makes the number of actual blocks indeterminate until execution.